### PR TITLE
Saturate list index on usize.

### PR DESCRIPTION
**Pull request by falsetru**

### CODE REVIEW

File: vm/src/sliceable.rs [hunk 1]
```diff
@@ -315,6 +315,19 @@ pub fn saturate_index(p: isize, len: usize) -> usize {
     p as usize
 }
 
+// Saturate p in range [0, len] inclusive
+pub fn pyint_saturate_index(p: PyIntRef, len: usize) -> usize {
```
Comment (youknowone) [inline vm/src/sliceable.rs:319]: The general approach is perfect, but I need time to think where will be the best place to put this function

File: vm/src/builtins/list.rs [hunk 2]
```diff
@@ -271,15 +271,25 @@ impl PyList {
...
+                pyint_saturate_index(int, len)
+            }
+        };
+        let stop: usize = match stop {
+            OptionalArg::Missing => len,
+            OptionalArg::Present(obj) => {
+                let int: PyIntRef = obj.try_into_value(vm)?;
+                pyint_saturate_index(int, len)
+            }
+        };
```
[the remark this answers is not in the document]
Comment (falsetru) [reply vm/src/builtins/list.rs]: Updated as you suggested, with minor change to make compiler happy.

[`d382e4a` (#3772)](https://github.com/RustPython/RustPython/pull/3772/commits/d382e4ad2937ea8f70072957871bf416a1136aa6)

**Comment by falsetru** [comment]
rebased to resolve conflict.

**Review by youknowone — APPROVED** [review]
Thank you for contributing!

**Pull request merged**